integration_test/common/common.go

```diff
@@ -121,3 +101,33 @@ func SliceContains(slice []string, toFind string) bool {
 	}
 	return false
 }
+
+func NewIntegrationMetadataValidator() *validator.Validate {
+	v := validator.New()
+	_ = v.RegisterValidation("onerepresentative", func(fl validator.FieldLevel) bool {
+
+		fieldStr := fl.Field().String()
+		if fieldStr == "" {
+			// Ignore the case where this field is not actually specified or is left empty.
+			return true
+		}
+
+		metrics, ok := fl.Field().Interface().([]*ExpectedMetric)
+		if !ok {
+			return false
+		}
+
+		if len(metrics) == 0 {
+			return true
+		}
+
+		representativeCount := 0
+		for _, m := range metrics {
+			if m.Representative {
+				representativeCount += 1
+			}
+		}
+		return representativeCount == 1
```

Comment:
Do we also need a `RegisterTranslation` call to define an error message for this validator?

Reply:
That was my initial thought too. The current error msg isn't clear on the root cause on why it fails the test, it only says it fails on the tag, thus the `onetrue` tag needs to have a comment saying on what this tag expected. Such comment might be better preserved in a well explained error message.

We have this in ops-agent validator. That is client facing thus we must explain it well while this is internal, maybe a comment might suffice if the effort on implementing the error message needs more than moderate effort.

Reply:
@avilevy18 and I had an offline discussion about it. Doing this via the translations mechanism turns out to be extremely cumbersome and counter-intuitive. We can do the same kind of switch as we have in the config validator, but that would have to be a follow-up change.